src/atoms/cc-badge.js

```diff
@@ -0,0 +1,149 @@
+import { css, html, LitElement } from 'lit-element';
+import { defaultThemeStyles } from '../styles/default-theme.js';
+
+/**
+ * @typedef {import('./types.js').Intent} Intent
+ * @typedef {import('./types.js').Weight} Weight
+ */
+
+/**
+ * A component to highlight a small chunk of text.
+ *
+ * @cssdisplay inline-flex
+ */
+export class CcBadge extends LitElement {
+  static get properties () {
+    return {
+      circle: { type: Boolean, reflect: true },
+      iconAlt: { type: String },
+      iconSrc: { type: String },
+      intent: { type: String, reflect: true },
+      text: { type: String },
+      weight: { type: String, reflect: true },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {Boolean} Sets the badge to a bubble style. Should only be used to display 1 or 2 digits figures. */
+    this.circle = false;
+
+    /** @type {String|null} Sets the `alt` attribute value on the `<img>` tag. Only use if the image conveys additional info compared to surrounding text. */
+    this.iconAlt = null;
+
+    /** @type {String|null} Sets the icon displayed on the left of the text inside the badge. */
+    this.iconSrc = null;
+
+    /** @type {Intent} Sets the accent color used for the badge. */
+    this.intent = 'neutral';
+
+    /** @type {String} Sets the text content of the badge. */
+    this.text = null;
+
+    /** @type {Weight} Sets the style of the badge depending on how much one wants it to stand out. */
+    this.weight = 'dimmed';
+  }
+
+  render () {
+    return html`
+      ${this.iconSrc != null ? html`
+        <img src=${this.iconSrc} alt=${this.iconAlt ?? ''}>
+      ` : ''}
+      <span>${this.text}</span>
+    `;
+  }
+
+  static get styles () {
+    return [
+      defaultThemeStyles,
+      // language=CSS
+      css`
+        :host {
+          align-items: center;
+          /* only used as fallback in case theme is not set */
+          background-color: #777;
+          border-radius: 1em;
+          box-sizing: border-box;
+          color: var(--color-text-default);
+          display: inline-flex;
+          font-size: 0.8em;
+          gap: 0.3em;
+          min-height: 1.7em;
+          padding: 0.2em 0.8em;
+        }
+
+        img {
+          height: 1em;
+        }
+
+        :host([circle]) {
+          border-radius: 50%;
+          font-size: 1em;
+          height: 1.5em;
+          justify-content: center;
+          min-height: unset;
+          padding: 0;
+          width: 1.5em;
+        }
+
+        :host([weight="dimmed"]) {
+          background-color: var(--cc-badge-accent-color);
+        }
+
+        :host([weight="strong"]) {
+          background-color: var(--cc-badge-accent-color);
+          color: var(--color-text-inverted);
+        }
+
+        :host([weight="outlined"]) {
+          background-color: var(--color-bg-default);
+          border: solid 0.1em var(--cc-badge-accent-color);
```

Comment:
To be honest it depends, in most cases I don't want the border to scale with the font size but in this case, the border is what makes the "outlined" badges stand out so I thought it was a good idea to make them scale so that if you increase font size, they would still stand out from the rest of the content.

You can compare below, with font size increase a lot (@400% on FF):
- 1px: ![image](https://user-images.githubusercontent.com/100240294/176388391-9a9c1225-8df2-464c-a0b3-05ce3d3ea2a8.png)
- 0.1em: 
![image](https://user-images.githubusercontent.com/100240294/176388512-86e365c2-5299-4fd8-a0da-2074e47b0b27.png)